### ISSUE TITLE
`consume` never returns if task function doesn't contain `produce`

### ISSUE THREAD

**Issue by yebai** [TuringLang/Libtask.jl#41]
The following code would halt

```julia

using Libtask

function f_ct()
   t = 0;
   while t<3
     println(t)
     t = 1 + t
   end
   return t
end

t = CTask(f_ct)

julia> consume(t) # Never returns. 
0
1
2

```

This is likely due to the `consumer` task never gets waken up if the `producer` task exit before any `produce` statement. Maybe a better behaviour for this kind of tasks is to wake up the `producer` task with the `producer` task's `return` value?

**Comment by devmotion** [comment]
As a comparison, the following happens with a `Channel`-based implementation:

The first consumer-producer example in the README could be implemented as
```julia
julia> chnl = Channel(; spawn=true) do ch
           t = 0
           while true
               put!(ch, t)
               t += 1
           end
       end;

julia> isempty(chnl)
false

julia> take!(chnl)
0

julia> take!(chnl)
1
```

The example in this issue gives
```julia
julia> chnl = Channel(; spawn=true) do ch
          t = 0
          while t < 3
            println(t)
            t = 1 + t
          end
          return t
       end;

julia> 0
1
2

julia> isempty(chnl)
true

julia> take!(chnl)
ERROR: InvalidStateException("Channel is closed.", :closed)
Stacktrace:
 [1] check_channel_state at ./channels.jl:167 [inlined]
 [2] take_unbuffered(::Channel{Any}) at ./channels.jl:403
 [3] take!(::Channel{Any}) at ./channels.jl:381
 [4] top-level scope at REPL[69]:1
```
I.e., the function runs once when the channel is constructed and the channel is closed afterwards.

**Comment by yebai** [comment]
Will be fixed by #93
